CONTRIBUTING.md

```diff
@@ -6,6 +6,212 @@ This guide explains how to contribute to cocotb, and documents the processes we
 All processes in this document are designed to streamline the development effort, to avoid bottlenecks, and to ultimately give a pleasant experience to all involved.
 
 
+## Getting Help
+
+Cocotb is a diverse and challenging project to contribute to.
+If you ever feel lost, out of your depth, or simply want to know more, the [cocotb gitter channel](https://gitter.im/cocotb/Lobby) is actively watched by many cocotb users, contributors, and maintainers.
+It is a good idea if you are unsure whether your problem or question is worthy of a Github Issue to first post it to the gitter channel.
+You may also ask questions in [Github issues](https://github.com/cocotb/cocotb/issues).
+If you don't receive any response on the gitter channel or a Github issue, or you want help privately, you may directly contact a [maintainer](#maintainers).
+
+
+## What to Work On
+
+There is *a lot* of work to do on this project, no matter your area of expertise or skill level.
+
+If you are a beginner there are several [Github issues](https://github.com/cocotb/cocotb/issues) marked [`good first issue`](https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) you can look at.
+There are also a number of things you can work on that aren't in Github issues and don't require in-depth knowledge of the cocotb internals.
+They including the following:
+
+* Documentation improvements
+* Managing Github issues and the gitter channel
+* Testing and coverage improvements
+
+Cocotb is still not perfect. There are plenty of [bug fixes](https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3Atype%3Abug) and [features])(https://github.com/cocotb/cocotb/issues?q=is%3Aopen+is%3Aissue+label%3Atype%3Afeature) that can be worked.
+Most of these are recorded as Github issues.
+
+
+### Documentation
+
+Cocotb's documentation is always open to improvements.
+Improving documentation will help users better understand and use cocotb;
+and may decrease the number of questions the gitter channel and Github issue page.
+Updating documentation requires the following:
+
+* A good grasp on English
```

Comment:
```suggestion
Updating documentation requires knowledge of:

```
Requiring "good" english might deter someone who feels unsure but is actually good enough. That one needs to have a certain level is obvious IMO.